File: blSystem-frontend/src/app/model/Request.ts
import { Donor } from "./Donor"
import { BloodGroup } from "./enums/BloodGroup"
import { RequestStatus } from "./enums/RequestStatus"

export class Request {
    id!: number
    recipientId!: number 
    requestDate!: Date
    bloodGroup!: BloodGroup
    isUrgent!: boolean
    status!: RequestStatus;

   
}

File: blSystem-backend/src/main/java/com/blsystem/entity/Request.java
package com.blsystem.entity;

import com.blsystem.entity.enums.BloodGroup;
import com.blsystem.entity.enums.RequestStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "requests")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Request {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long recipientId;
    private String requestDate;
    @Enumerated(EnumType.STRING)
    private BloodGroup bloodGroup;
    private boolean isUrgent;
    @Enumerated(EnumType.STRING)
    private RequestStatus status;

}
